components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/ServiceProviderToApiModel.java

```diff
@@ -489,6 +493,28 @@ private AdvancedApplicationConfiguration buildAdvancedAppConfiguration(ServicePr
                         .getAndroidAttestationServiceCredentials()));
     }
 
+    /**
+     * Retrieves the trusted app configuration for an application's advanced configuration based on the provided
+     * service provider.
+     *
+     * @param serviceProvider The service provider for which trusted app configuration is required.
+     * @return An instance of TrustedAppConfiguration containing trusted app configurations of the application.
+     */
+    private TrustedAppConfiguration getTrustedAppConfiguration(ServiceProvider serviceProvider) {
+
+        SpTrustedAppMetadata trustedAppMetadata = serviceProvider.getTrustedAppMetadata();
+        if (trustedAppMetadata == null) {
+            trustedAppMetadata = new SpTrustedAppMetadata();
+        }
+
+        return new TrustedAppConfiguration()
+                .isFIDOTrustedApp(trustedAppMetadata.getIsFidoTrusted())
+                .androidPackageName(trustedAppMetadata.getAndroidPackageName())
+                .androidThumbprints(Arrays.asList(trustedAppMetadata.getAndroidThumbprints()
+                        .split(ATTRIBUTE_SEPARATOR)))
```

Comment:
Lets have the thumprints as an array in the model so we can avoid this sort of splitting in multiple places.